opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/samplers/ProbabilityProportionalToSizeAbstractQuerySampler.java

```diff
@@ -137,7 +135,10 @@ public String sample() throws Exception {
         // This may require generating more random numbers than what was requested
         // because removing duplicate user queries will require randomly picking more queries.
         int count = 1;
-        while(querySet.size() < parameters.getQuerySetSize() && count < userQueries.size()) {
+
+        // TODO: How to short-circuit this such that if the same query gets picked over and over, the loop will never end.
```

Comment:
The issue of short-circuiting the PPTSS sampling for when the same user query gets picked over and over and over...